ibm/resource_ibm_container_vpc_cluster_worker_volume_attachment.go

```diff
@@ -0,0 +1,274 @@
+		return err
+	}
+
+	workersAPI := wpClient.Workers()
+	target, err := getVpcClusterTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	clusterNameorID := d.Get("cluster").(string)
+	volumeAttachmentID := d.Id()
+	workerID := d.Get("worker").(string)
+
+	volume, err := workersAPI.GetStorageAttachment(clusterNameorID, workerID, volumeAttachmentID, target)
+	if err != nil {
+		return err
+	}
+	d.Set("volume_attachment_name", volume.Name)
+	d.Set("status", volume.Status)
+	d.Set("volume_type", volume.Type)
+	return nil
+}
+
+func resourceIBMContainerVpcWorkerVolumeAttachmentDelete(d *schema.ResourceData, meta interface{}) error {
+	wpClient, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	workersAPI := wpClient.Workers()
+	target, err := getVpcClusterTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	clusterNameorID := d.Get("cluster").(string)
+	volumeAttachmentID := d.Id()
+	workerID := d.Get("worker").(string)
+
+	detachVolumeRequest := v2.VolumeRequest{
+		Cluster:            clusterNameorID,
+		VolumeAttachmentID: volumeAttachmentID,
+		Worker:             workerID,
+	}
+
+	_, Err := workersAPI.DeleteStorageAttachment(detachVolumeRequest, target)
+	if err != nil {
+		return fmt.Errorf("Failed to update the the volume attachment to different worker node: %s", Err)
+	}
+	return nil
+}
+
+func resourceIBMContainerVpcWorkerVolumeAttachmentExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	wpClient, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return false, err
+	}
+
+	workersAPI := wpClient.Workers()
+	target, err := getVpcClusterTargetHeader(d, meta)
+	if err != nil {
+		return false, err
+	}
+
+	clusterNameorID := d.Get("cluster").(string)
+	volumeAttachmentID := d.Id()
+	workerID := d.Get("worker").(string)
+
+	_, Err := workersAPI.GetStorageAttachment(clusterNameorID, workerID, volumeAttachmentID, target)
+	if Err != nil {
+		if apiErr, ok := Err.(bmxerror.RequestFailure); ok {
+			if apiErr.StatusCode() == 404 {
+				return false, nil
+			}
+		}
+		return false, fmt.Errorf("Error communicating with the API: %s", Err)
+	}
+	return true, nil
+}
+
+func waitforVolumetoAttach(d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	wpClient, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return nil, err
+	}
+
+	workersAPI := wpClient.Workers()
+
+	target, trgetErr := getVpcClusterTargetHeader(d, meta)
+	if trgetErr != nil {
+		return nil, trgetErr
+	}
+	clusterNameorID := d.Get("cluster").(string)
+	volumeAttachmentID := d.Id()
+	workerID := d.Get("worker").(string)
+
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{volumeAttaching},
+		Target:  []string{volumeAttached},
+		Refresh: func() (interface{}, string, error) {
+			volume, Err := workersAPI.GetStorageAttachment(clusterNameorID, workerID, volumeAttachmentID, target)
+
+			if Err != nil {
+				return volume, deployInProgress, Err
```

Comment:
where we declated deployInProgress i think that shd be also in one of the Target